bcgov/PIMS
# PIMS-1946 Cancel/Stop Notifications

## 🎯 Summary

  
[PIMS-1946](https://citz-imb.atlassian.net/browse/PIMS-1946)


Tackles two issues with the notifications:
- Notifications previously queued when a project was cancelled were still being sent.
- Notifications were being re-sent when a project returned to a previous status.

## Changes
- If a project is cancelled, Pending and Accepted notifications will be cancelled.
- If a project's status is changed, the notifications are only queued if this status is not in the project's history.

## Testing
- Alter the status of a project in and out of the statuses that queue notifications (Approved for ERP, Submitted)
  - See that repeat notifications should not be queued.
- Change the agency interest values
  - See that the interest values should be updated still
- Cancel a project with pending notifications
  - See that those notifications are now cancelled too


## 🔰 Checklist


> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.
